core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -129,14 +134,15 @@ public T acquire(Integer time, TimeUnit unit) {
         mTakeLock.unlock();
       }
     } catch (InterruptedException e) {
-      throw new RuntimeException(e);
+      throw new IOException(e);
```

Comment:
System code is fine, but you can't guarantee the customized client code behavior. In fact, a previous uncommitted version is based on throws RuntimeException, and ```public T acquire(long time, TimeUnit unit) throws Exception``` (not IO). Since thrown Exception is to general, it has some bad effects on the upper codes.